src/Estimators/ScalarEstimatorInputs.h

```diff
@@ -45,19 +80,58 @@ class CSLocalEnergyInput
     CSLocalEnergyInputSection()
     {
       section_name = "CSLocalEnergy";
-      attributes   = {"npsi", "type"};
+      attributes   = {"name", "npsi", "type"};
       integers     = {"npsi"};
-      strings      = {"type"};
+      strings      = {"name", "type"};
     }
   };
 
 public:
+  using Consumer       = CSEnergyEstimator;
+  CSLocalEnergyInput() = default;
   CSLocalEnergyInput(xmlNodePtr cur);
   int get_n_psi() const { return n_psi_; }
+  const std::string& get_name() const { return name_; }
+  const std::string& get_type() const { return type_; }
 
 private:
+  std::string name_;
+  std::string type_;
   CSLocalEnergyInputSection input_section_;
   int n_psi_ = 1;
 };
 
+class RMCLocalEnergyEstimator;
+
+class RMCLocalEnergyInput
+{
+  class RMCLocalEnergyInputSection : public InputSection
+  {
+  public:
+    RMCLocalEnergyInputSection()
+    {
+      section_name = "RMCLocalEnergy";
+      attributes   = {"name", "npsi", "type"};
+      integers     = {"npsi"};
+      strings      = {"name", "type"};
+    }
+  };
+
+public:
+  using Consumer        = RMCLocalEnergyEstimator;
+  RMCLocalEnergyInput() = default;
+  RMCLocalEnergyInput(xmlNodePtr cur);
+  int get_n_psi() const { return n_psi_; }
+  const std::string& get_name() const { return name_; }
+  const std::string& get_type() const { return type_; }
+
+private:
+  std::string name_;
+  std::string type_;
+  RMCLocalEnergyInputSection input_section_;
+  int n_psi_ = 1;
```

Comment:
I cannot find any document or use of n_psi_, could you delete that in the whole code base?

Reply:
I didn't finish up for RMC which I should have. For RMC it should actually be n_obs_ and it will be used when these are used to construct the actual scalar estimators.

Reply:
Then write down what your learnt for now.